src/main/java/org/candlepin/subscriptions/tally/filler/TallyReportDataPointAdapter.java

```diff
@@ -35,12 +42,13 @@ public boolean itemIsLarger(TallyReportDataPoint oldData, TallyReportDataPoint n
   public TallyReportDataPoint createDefaultItem(
       OffsetDateTime itemDate, TallyReportDataPoint previous, boolean useRunningTotalFormat) {
     var point = new TallyReportDataPoint().date(itemDate).hasData(false).value(0);
-    if (useRunningTotalFormat) {
-      // Guard against previous being null and against a previous object with a value field equal
-      // to null
-      var prevValue = (previous == null) ? 0 : Objects.requireNonNullElse(previous.getValue(), 0);
-      point.value(prevValue);
+    // itemDate is already adjusted in ReportFiller to be at the period start
+    if (itemDate.isBefore(clock.now()) && useRunningTotalFormat && previous != null) {
+      point
+          .hasData(previous.getValue() != null)
+          .value(Objects.requireNonNullElse(previous.getValue(), 0));
     }
+
     return point;
```

Comment:
How does this behave if we have a gap in the data, especially if a customer turns off systems for the weekend, and then comes back, do we fill in the gaps until `now()` or not? If there's not a gap, I would appreciate some code comments explaining how this behaves in that scenario.

Reply:
There shouldn't be a gap.  The `testTallyReportDataFillerMarksHasDataFalseAfterNowWithRunningTotal` unit test covers this I believe:

```java
List<TallySnapshot> snapshots =  List.of(1, 2, 8).stream()...
```

And then ensures there are records for every day requested.  The `ReportFiller.fillWithRange()` method is what handles making sure we tick through every period in the range requested.  I can add a comment to that effect.